spring-cloud-gcp-bigquery/src/main/java/com/google/cloud/spring/bigquery/core/BigQueryTemplate.java

```diff
@@ -124,16 +125,26 @@ public void setJobPollInterval(Duration jobPollInterval) {
 	@Override
 	public ListenableFuture<Job> writeDataToTable(
 			String tableName, InputStream inputStream, FormatOptions dataFormatOptions) {
+		return this.writeDataToTable(tableName, inputStream, dataFormatOptions, null);
+	}
+
+	@Override
+	public ListenableFuture<Job> writeDataToTable(
+			String tableName, InputStream inputStream, FormatOptions dataFormatOptions, Schema schema) {
+
 		TableId tableId = TableId.of(datasetName, tableName);
 
-		WriteChannelConfiguration writeChannelConfiguration = WriteChannelConfiguration
+		WriteChannelConfiguration.Builder writeChannelConfiguration = WriteChannelConfiguration
 				.newBuilder(tableId)
 				.setFormatOptions(dataFormatOptions)
-				.setAutodetect(this.autoDetectSchema)
 				.setWriteDisposition(this.writeDisposition)
-				.build();
+				.setAutodetect(this.autoDetectSchema);
 
-		TableDataWriteChannel writer = bigQuery.writer(writeChannelConfiguration);
+		if (schema != null) {
```

Comment:
Why allow `null` if there is a version of the method that skips this parameter?

Reply:
Yes I added it because I wanted to delegate to this method from `ListenableFuture<Job> writeDataToTable(String tableName, InputStream inputStream, FormatOptions dataFormatOptions);`